### DISCUSSION TITLE
Confusion of staticcheck/gosimple/stylecheck/unused linters and config file

### DISCUSSION THREAD

**Discussion by arxeiss** [golangci/golangci-lint#1989] [Q&A] [answered]
I was just wanted to open a bug report, when I noticed this PR https://github.com/golangci/golangci-lint/pull/1946

So if I understand it correctly. Linters `staticcheck`, `gosimple`, `stylecheck` and `unused` are running the same tool known as https://staticcheck.io/ ? If do so, why it is there 4 times?

And why actually the staticcheck config file `staticcheck.conf` is ignored when running with golangci-lint?

**Comment by ldez** [comment, marked as answer]
> If do so, why it is there 4 times?

staticcheck is a binary and a set of rules, those rules are grouped in "category": `staticcheck`, `gosimple`, `stylecheck` and `unused`.

- staticcheck: https://github.com/dominikh/go-tools/blob/331bcca40621580531c9b8a9cb7d5c2c06c709d2/staticcheck/analysis.go#L23
- stylecheck: https://github.com/dominikh/go-tools/blob/331bcca40621580531c9b8a9cb7d5c2c06c709d2/stylecheck/analysis.go#L14
- simple: https://github.com/dominikh/go-tools/blob/331bcca40621580531c9b8a9cb7d5c2c06c709d2/simple/analysis.go#L11
- unused: https://github.com/dominikh/go-tools/blob/331bcca40621580531c9b8a9cb7d5c2c06c709d2/unused/unused.go#L418

megacheck was the first name of staticcheck, the "categories" were binaries, and as you can understand there are historical reasons behind those "linters" inside golangci-lint.

More information about the 4 linters: https://github.com/golangci/golangci-lint/issues/357

---

> And why actually the staticcheck config file staticcheck.conf is ignored when running with golangci-lint?

The file is partially ignored: you can use `staticcheck.conf` to define `initialisms`, `dot_import_whitelist`, `http_status_code_whitelist` but `checks` are ignored (instead you have to use `issues.excludes` in the golanci-lint configuration).

It is mainly related to the API of staticcheck: we don't have access to the element to handle the staticcheck configuration.

About the configuration, you can follow https://github.com/golangci/golangci-lint/issues/1275

**Reply by arxeiss** [reply]
Thank you for clarifying..

**Reply by ldez** [reply]
Update: golangci-lint now support staticcheck configuration https://github.com/golangci/golangci-lint/pull/2017

Note that the file `staticcheck.conf` will not be read after this PR.
